doc/decisions/4_in_review/change_tracking.md

```diff
@@ -174,13 +174,26 @@ The changetracking plugin needs to export at least functions for the following t
 
 ## Decision
 
+Plugins and application developer can enable changetracking via a contract.
```

Comment:
I don't really see the use case for this. Important is that admins can enable change tracking via `kdb`.

Reply:
I don't see it would work from the technical POV for a plugin to enable/disable changetracking.

I also agree that it is most important that it can be enabled/disabled via `kdb`, i.e. that there is a config key somewhere in `/elektra`.

Reply:
I don't think disabling via `/elektra` is a good way. We're talking about change tracking, NOT recording here. Some plugins might depend on changetracking, i.e. the notification plugins. 

Wouldn't it be possible that plugins export, i.e. `system:/elektra/modules/<pluginname>/requirechangetracking` in their contract?

Reply:
Then you need to define what exactly "enabling changetracking" does. Also is there even a use case for disabling it? Shouldn't it just be automatically active whenever it's needed? Because than I'd say it is not about "enabling changetracking", but about "declaring that changetracking is required".

Reply:
Enabling changetracking means we fill the additional keyset in `struct _KDB` with deep-duped keys. If it is not enabled, the keyset will not be allocated.

Reply:
It's just an optimization - if no plugin requires it, why do it? Yes I probably should change the wording from "enabling" to "requiring"